### ISSUE TITLE
Original request method

### ISSUE THREAD

**Issue by sarneeh** [elbywan/wretch#35]
So I'd like to create an app-wide `wretch` instance which will retry requests on `unauthorized` statuses. It would be pretty easy with `wretch`, but unfortunately, I don't really know how do I replay the same specific request.

Because actually sending the request requires a call to `.get()`, `.post()` or whatever method it was, I can't replay the same request having a global instance.

Am I missing something? Is there a way to send the request without calling the `.[httpMethod]()` function?

**Comment by sarneeh** [comment]
@elbywan Ehm... You've posted a link to my issue 😄

**Comment by elbywan** [comment]
@sarneeh Oops my bad 😄!

Here is the correct [link](https://github.com/elbywan/wretch/issues/15)!

**Comment by sarneeh** [comment]
I've seen this particular issue and the documentation and they still don't answer my question.

Let me show you an example based on your post:

```js
const reAuthOn401 = wretch()
  .catcher(401, async (error, originalRequest) => {
    const token = await renewToken()
    return originalRequest
      .auth(`Bearer ${ token }`)
      .get() // This won't work as my originalRequest was a .post()
      .unauthorized(err => { throw err })
      .json()
  })

reAuthOn401.url("/resource")
  .post({})
  .json()
```

**Comment by elbywan** [comment]
> I've seen this particular issue and the documentation and they still don't answer my question. 

I'm sorry I  understand the issue now.

There is indeed a way to achieve this, event if it may (rightfully) seem a bit hacky but that should still work.

```js
// Same example

const reAuthOn401 = wretch()
  .catcher(401, async (error, originalRequest) => {
    const token = await renewToken()
    // Reuse the same http method
    const method = originalRequest._options.method.toLowerCase()
    return originalRequest
      .auth(`Bearer ${ token }`)
      [method]()
      .unauthorized(err => { throw err })
      .json()
  })

reAuthOn401.url("/resource")
  .post({})
  .json()
```

I really need to think about this. Maybe I'll expose this variable in a less hacky way, or at least document the `_options` field.

In the meantime if is safe to use the syntax above anyway.

**Comment by sarneeh** [comment]
@elbywan It would be great if the `originalRequest` could have some additional API for this kind of situation. This would make this library even more awesome! 😄 Thank you very much for the example. I'd love to help to implement this but unfortunately, I'm completely out of time for about a month right now 😢 But if you don't have it either you can leave me this issue for later resolution 😄

**Comment by elbywan** [comment]
>I'd love to help to implement this but unfortunately, I'm completely out of time for about a month right now 😢 But if you don't have it either you can leave me this issue for later resolution 😄

Sure, thanks! I'm also a bit busy these days but I think I can free some time before a month to work on this. If so I'll update this issue 😉.

**Comment by MarlBurroW** [comment]
Any news about this issue ? I'm exactly in the same case (global api instance with a global 401 catcher for multiple methods), this is a very common use case and didn't find an elegant way to replay the original request (with the same method). 

I think something like this could be nice:

```javascript
// Same example

const reAuthOn401 = wretch()
  .catcher(401, async (error, originalRequest) => {
    const token = await renewToken()

    const method = originalRequest._options.method.toLowerCase()
    return originalRequest .auth(`Bearer ${ token }`).play()  // Replay the request with original method then go ... (see below)         
})

reAuthOn401.url("/resource")
  .post({}) 
  // ... Here !
  .json()
```
But I do not know if the Fetch or Promise API allows to do this.

**Comment by sarneeh** [comment]
@MarlBurroW Unfortunately I'm still completely out of time, but I guess PRs are still welcome :smile:

**Comment by elbywan** [comment]
@sarneeh @MarlBurroW Sorry for the delay, I was quite busy also 😉 .

I just added a new `.replay()` method that should do the trick. (released in `1.5.0`)

**Comment by sarneeh** [comment]
@elbywan Awesome, thank you! :smile:
